tests/Unit/Helpers/Tests/Test_MakeRandomVectorInMagnitudeRange.cpp

```diff
@@ -0,0 +1,149 @@
+
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeRandomVectorInMagnitudeRange.hpp"
+
+template <typename DataType, size_t Dim, typename Fr = Frame::Inertial>
+tnsr::ii<DataType, Dim, Fr> random_spatial_metric(
+    const gsl::not_null<std::mt19937*> generator,
+    const DataType& used_for_size) noexcept {
+  std::uniform_real_distribution<> distribution(-0.05, 0.05);
+  auto spatial_metric = make_with_random_values<tnsr::ii<DataType, Dim, Fr>>(
+      generator, make_not_null(&distribution), used_for_size);
+  for (size_t d = 0; d < Dim; ++d) {
+    spatial_metric.get(d, d) += 1.0;
+  }
+  return spatial_metric;
+}
+
+void check_random_values(
+    const gsl::not_null<std::unordered_set<double>*> values,
+    const gsl::not_null<size_t*> counter, const double v,
+    const double lower_bound, const double upper_bound) noexcept {
+  CHECK(v >= lower_bound);
+  CHECK(v <= upper_bound);
+  CHECK(values->insert(v).second);
+  ++(*counter);
+}
+
+// clang-tidy is wrong, this is a function definition
+template <typename T>
+void check_random_values(
+    const gsl::not_null<std::unordered_set<double>*> values,        // NOLINT
+    const gsl::not_null<size_t*> counter, const T& c,               // NOLINT
+    const double lower_bound, const double upper_bound) noexcept {  // NOLINT
+  for (const auto& v : c) {
+    check_random_values(values, counter, v, lower_bound, upper_bound);
+  }
+}
+
+template <typename... Tags>
+void check_random_values(
+    const gsl::not_null<std::unordered_set<double>*> values,
+    const gsl::not_null<size_t*> counter,
+    const Variables<tmpl::list<Tags...>>& v, const double lower_bound,
+    const double upper_bound) noexcept {
+  expand_pack((check_random_values<decltype(get<Tags>(v))>(
+                   values, counter, get<Tags>(v), lower_bound, upper_bound),
+               cpp17::void_type{})...);
+}
+
+// Compute Magnitude
+// These functions are a workaround using Requires to
+// figure out if the inverse metric is needed
+template <typename DataType, size_t Dim, UpLo Ul, typename Fr,
+          Requires<(Ul == UpLo::Up)> = nullptr>
+Scalar<DataType> magnitude_auto_invert(
+    const Tensor<DataType, Symmetry<1>, index_list<SpatialIndex<Dim, Ul, Fr>>>&
+        vector,
+    const tnsr::ii<DataType, Dim, Fr>& metric) {
+  return magnitude(vector, metric);
+}
+
+template <typename DataType, size_t Dim, UpLo Ul, typename Fr,
+          Requires<(Ul == UpLo::Lo)> = nullptr>
+Scalar<DataType> magnitude_auto_invert(
+    const Tensor<DataType, Symmetry<1>, index_list<SpatialIndex<Dim, Ul, Fr>>>&
+        vector,
+    const tnsr::ii<DataType, Dim, Fr>& metric) {
+  const tnsr::II<DataType, Dim, Fr> inverse_metric =
+      determinant_and_inverse(metric).second;
+
+  return magnitude(vector, inverse_metric);
+}
+
+// Test copied from Test_RandomUnitNormal
```

Comment:
I don't see anything in `Test_RandomUnitNormal.cpp` that looks like this.

Reply:
I removed the comment. Seems the implementation started deviating to much.